backend/copr_backend/helpers.py

```diff
@@ -627,7 +628,49 @@ def get_chroot_arch(chroot):
     return chroot.rsplit("-", 2)[2]
 
 
-def pkg_name_evr(srpm_path, logger):
+def find_spec_file(folder, logger):
+    """
+    Returns spec file path or None
+    """
+    spec_pattern = os.path.join(folder, "*.spec")
+    result = glob.glob(spec_pattern)
+    if not result:
+        return None
+
+    if len(result) > 1:
+        logger.error("More than one spec file found! Using the first %s from %s",
+                     str(result[0]), str(result))
+
+    return result[0]
+
+
+def format_evr(epoch, version, release):
+    """
+    Return evr in format (epoch:)version-release
+    """
+    if epoch is not None and epoch.isdigit():
+        return f"{epoch}:{version}-{release}"
+
+    return f"{version}-{release}"
+
+
+def pkg_name_evr(spec_file_path, srpm_path, logger):
+    """
+    Queries a SRPM or spec file for its name and evr (epoch:version-release)
+    """
+    if spec_file_path is None:
+        return pkg_name_evr_from_srpm(srpm_path, logger)
+
+    try:
+        spec = Specfile(spec_file_path)
+    except Exception as exc:
+        logger.error("Exception appeared during handling spec file: %s", str(exc))
+        raise CoprBackendSrpmError(str(exc)) from exc
```

Comment:
Small wish, can we reverse this logic a bit and - if Specfile parsing fails - fallback to `pkg_name_evr_from_srpm` as we did before this PR?

Reply:
I am afraid I don't get it here. Do you mean by reversing the logic - if Specfile parsing fails, then call the `pkg_name_evr_from_srpm`?